nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -2098,31 +2098,24 @@ void ddxUseMsg()
 
 static int nxagentGetDialogName()
 {
-  snprintf(nxagentDialogName, NXAGENTDIALOGNAMELENGTH, "NX");
-
   if (*nxagentSessionId != '\0')
   {
     int length = strlen(nxagentSessionId);
 
-    strcpy(nxagentDialogName, "NX - ");
-
     /* if the session id contains an MD5 hash in a well-known format cut it off */
     if (length > (MD5_LENGTH * 2 + 1) &&
            *(nxagentSessionId + (length - (MD5_LENGTH * 2 + 1))) == '-')
     {
-      strncat(nxagentDialogName, nxagentSessionId,
-              MIN(NXAGENTDIALOGNAMELENGTH - strlen(nxagentDialogName), length - (MD5_LENGTH * 2 + 1)) - 1);
-    }
-    else
-    {
-      strncat(nxagentDialogName, nxagentSessionId, NXAGENTDIALOGNAMELENGTH - strlen(nxagentDialogName) - 1);
+      length -= (MD5_LENGTH * 2 + 1);
     }
 
-    nxagentDialogName[NXAGENTDIALOGNAMELENGTH - 1] = '\0';
+    snprintf(nxagentDialogName, NXAGENTDIALOGNAMELENGTH, "NX - %.*s", length, nxagentSessionId);
```

Comment:
Ping on that?

Reply:
ok, must think about that one, but not now.

Reply:
`2*MD5 + 1` actually... and yes, that's fine. I got confused by the different behaviors regarding the total string length (since `snprintf` expects the full length, but the `str*` family normally the length excluding a terminating `NULL` character). That's not relevant here, so the code is fine. My bad.